Refactored the socket code to use Tokio for async I/O

Skywardbeta opened the pull request:
Refactored the socket code to use Tokio for async I/O while keeping socket2 for advanced configuration. Introduced a global Tokio runtime, made sockets non-blocking, and preserved the synchronous trait interface for smooth integration.

Skywardbeta commented:
Refactored the application to store all data in a ChatModel that notifies observers (like ChatView) of new messages, while ephemeral TCP/UDP sockets feed incoming data into the model, ensuring the UI remains updated

olivier-dj commented on src/app.rs:
```diff
@@ -1,11 +1,149 @@
+        app.sort_messages();
+    }
+
+    pub fn receive_message(&mut self, text: &str, sender: SharedPeer) {
```
Later, the msg sender is probably part of protobuf (a todo for later: remove this arg here)

olivier-dj commented on src/app.rs:
```diff
@@ -1,11 +1,149 @@
+        self.add_message(msg);
+    }
+
+    pub fn sort_messages(&mut self, ctx_peer_uuid: &str) {
```
oh.. so app.sort_message call this back? we had a ping pong call in send_message?

Skywardbeta commented:
- removed the logic that overwrote the message text on error, and ``send_message ``now takes a ``&mut Message`` rather than referencing UI fields directly.

- moved sorting logic into ``add_message ``and made sure the Message struct is pub avoiding ping-pong recursion and ensuring cleaner separation
